src/python/grpcio/grpc/_cython/_cygrpc/arguments.pyx.pxi

```diff
@@ -52,7 +52,9 @@ cdef grpc_arg _unwrap_grpc_arg(tuple wrapped_arg):
 
 cdef class _ArgumentProcessor:
 
-  cdef void c(self, argument, grpc_arg_pointer_vtable *vtable, references):
+  cdef void c(self, argument, grpc_arg_pointer_vtable *vtable, references) except *:
+    if not isinstance(argument, tuple) or len(argument) != 2:
```

Comment:
Do we need this when `except *` is present? Wouldn't we get an exception anyway?

If so, then I prefer not doing a manual typecheck, as it is unpythonic to unnecessarily restrict types that would just work fine here.

Reply:
The Cython exception handling is weird. I tried alternative ways without adding "except *", the TypeError exception will always crash the process with "Segment Fault".

Reply:
In that case, I recommend lifting that explicit typecheck into the Python layer, when the argument is passed to `Channel` or `Server` instead of doing it deep in Cython layer.

Reply:
Reading your comment again, I think I should clarify: I was suggesting adding `except *` but not throwing `TypeError` manually. Instead, relying on the actual function that iterates through the tuple to raise `ValueError` (since it obviously does recognize that as `ValueError` and prints the eaten exception to console). I think you may have to mark all the functions in the call stack as `except *` if the deeper function is the one raising `ValueError`.